core/heap.c

```diff
@@ -1517,7 +1517,7 @@ schedule_low_on_memory_event_trigger()
 /* Reserve virtual address space without committing swap space for it */
 static vm_addr_t
 vmm_heap_reserve(size_t size, heap_error_code_t *error_code, bool executable,
-                 which_vmm_t which)
+                 which_vmm_t which, bool reachable)
```

Comment:
This parameter shouldn't be necessary: instead test `which` for `VMM_REACHABLE`.